tests/FSharp.Compiler.ComponentTests/EmittedIL/StaticOptimizations/String_Enum.fs.il.bsl

```diff
@@ -0,0 +1,543 @@
+      .field public specialname rtspecialname int32 value__
+      .field public static literal valuetype assembly/String/Int32Enum Int32 = int32(0x00000001)
+    } 
+
+    .class auto ansi serializable sealed nested public Int64Enum
+           extends [runtime]System.Enum
+    {
+      .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+      .field public specialname rtspecialname int64 value__
+      .field public static literal valuetype assembly/String/Int64Enum Int64 = int64(0x1)
+    } 
+
+    .class auto ansi serializable sealed nested public ByteEnum
+           extends [runtime]System.Enum
+    {
+      .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+      .field public specialname rtspecialname uint8 value__
+      .field public static literal valuetype assembly/String/ByteEnum Byte = uint8(0x01)
+    } 
+
+    .class auto ansi serializable sealed nested public UInt16Enum
+           extends [runtime]System.Enum
+    {
+      .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+      .field public specialname rtspecialname uint16 value__
+      .field public static literal valuetype assembly/String/UInt16Enum UInt16 = uint16(0x0001)
+    } 
+
+    .class auto ansi serializable sealed nested public UInt32Enum
+           extends [runtime]System.Enum
+    {
+      .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+      .field public specialname rtspecialname uint32 value__
+      .field public static literal valuetype assembly/String/UInt32Enum UInt32 = uint32(0x00000001)
+    } 
+
+    .class auto ansi serializable sealed nested public UInt64Enum
+           extends [runtime]System.Enum
+    {
+      .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+      .field public specialname rtspecialname uint64 value__
+      .field public static literal valuetype assembly/String/UInt64Enum UInt64 = uint64(0x1)
+    } 
+
+    .method public static string  'string<CharEnum>'(valuetype assembly/String/CharEnum 'enum') cil managed
+    {
+      
+      .maxstack  3
+      .locals init (valuetype assembly/String/CharEnum V_0)
+      IL_0000:  ldarg.0
+      IL_0001:  stloc.0
+      IL_0002:  ldloca.s   V_0
```

Comment:
I have not looked into why this unnecessary defensive copy is made here for enums, but it seems likely to be related to the usage of `'T` in `let x = (# "" value : 'T #) in x.ToString()`, since it was already present for enums based on unsigned types before the new optimization for enums based on signed types was added:

https://github.com/dotnet/fsharp/blob/2be320198364322a7fd91934bd1d8ff0c255fc0b/tests/FSharp.Compiler.ComponentTests/EmittedIL/StaticOptimizations/String_Enum.fs.il.bsl#L117-L128